job-service/src/main/java/com/hpe/caf/services/job/api/JobsPut.java

```diff
@@ -224,7 +225,7 @@ public static String createOrUpdateJob(final String partitionId, String jobId, N
                 databaseHelper.reportFailure(partitionId, jobId, mapper.writeValueAsString(f));
 
                 //  Throw error message to user.
-                throw new Exception("Failed to add task data to the queue", ex);
+                throw new ServiceUnavailableException("Failed to add task data to the queue", ex);
```

Comment:
Should this be a 503? I think this usually happens when we forget to configure the job service with some queue ENV var it needs

Reply:
I went with 503 as we are catching only `IOException | TimeoutException ` at that level. For any missing or invalid queue data, I understand that a BadRequestException (400) is thrown `//  Make sure the target queue name is not empty. Null targetPipe is valid
            if ("".equals(jobTask.getTargetPipe())) {
                LOG.error("createOrUpdateJob: Error - '{}'", ERR_MSG_TARGET_QUEUE_NOT_SPECIFIED);
                throw new BadRequestException(ERR_MSG_TARGET_QUEUE_NOT_SPECIFIED);
            }

            //  Make sure the task queue name has been provided.
            if (!ApiServiceUtil.isNotNullOrEmpty(jobTask.getTaskPipe())) {
                LOG.error("createOrUpdateJob: Error - '{}'", ERR_MSG_TASK_QUEUE_NOT_SPECIFIED);
                throw new BadRequestException(ERR_MSG_TASK_QUEUE_NOT_SPECIFIED);
            }`